Assets/Plugins/Android/Core/AndroidFileIOHelper.cs

```diff
@@ -23,35 +23,50 @@
 namespace PlayEveryWare.EpicOnlineServices
 {
     using System;
+    using System.Threading.Tasks;
     using UnityEngine;
+    using UnityEngine.Networking;
 
-    public class AndroidFileIOHelper : MonoBehaviour
+    public class AndroidFileIOHelper
     {
-        public static string ReadAllText(string filePath)
+        public static async Task<string> ReadAllText(string filePath)
         {
-            using (var request = UnityEngine.Networking.UnityWebRequest.Get(filePath))
+            using UnityWebRequest request = UnityWebRequest.Get(filePath);
+            request.timeout = 2; //seconds till timeout
+            UnityWebRequestAsyncOperation operation = request.SendWebRequest();
+
+            while (!operation.isDone)
             {
-                request.timeout = 2; //seconds till timeout
-                request.SendWebRequest();
+                await Task.Yield();
+            }
 
-                //Wait till webRequest completed
-                while (!request.isDone) { }
+            string text = null;
 
-#if UNITY_2020_1_OR_NEWER
-                if (request.result != UnityEngine.Networking.UnityWebRequest.Result.Success)
-                {
-                    Debug.Log("Requesting " + filePath + ", please make sure it exists and is a valid config");
-                    throw new Exception("UnityWebRequest didn't succeed, Result : " + request.result);
-                }
-#else
-            if (request.isNetworkError || request.isHttpError)
+            switch (request.result)
             {
-                Debug.Log("Requesting " + filePath + ", please make sure it exists and is a valid config");
-                throw new Exception("UnityWebRequest didn't succeed : Network or HTTP Error");
-            }
-#endif
-                return request.downloadHandler.text;
+                case UnityWebRequest.Result.InProgress:
+                    // This should not happen, because of the Task.Yield() call
+                    // above.
+                    Debug.LogWarning("AndroidFileIOHelper: For some reason " +
+                                     "the request is still in progress.");
+                    break;
+                case UnityWebRequest.Result.ConnectionError:
+                case UnityWebRequest.Result.ProtocolError:
+                case UnityWebRequest.Result.DataProcessingError:
+                    Debug.LogError($"AndroidFileIOHelper: " +
+                                   $"UnityWebRequest for path " +
+                                   $"\"{filePath},\" failed with error code " +
+                                   $"'{request.result},' and error, " +
+                                   $"\"{request.error}.\"");
+                    break;
+                case UnityWebRequest.Result.Success:
+                    text = request.downloadHandler.text;
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException();
```

Comment:
I feel like we could log out the result/error here, assuming this can ever be hit.

Reply:
It cannot be hit right now - but is there because it completes the coverage of all possible cases within the `switch` statement.

Reply:
Just in case, can you please add a log that prints out the resulting error?

Reply:
Yup :) just did.